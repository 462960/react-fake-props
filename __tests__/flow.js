import path from 'path'
import fakeProps from '../src'

// const ComponentFile = path.join(__dirname, '../fixtures/ComponentWithPropTypes.jsx')
const SimpleFile = path.join(__dirname, '../fixtures/flow/Simple.jsx')

describe('fakeProps', () => {
  // it('should return an object with required props faked (snapshot)', () => {
  //   expect(fakeProps(ComponentFile)).toMatchSnapshot()
  // })
  //
  // it('should return an object with all props faked (snapshot)', () => {
  //   expect(fakeProps(ComponentFile, { optional: true })).toMatchSnapshot()
  // })
  //
  // it('should return more props with optional', () => {
  //   const allProps = fakeProps(ComponentFile, { optional: true })
  //   const requiredProps = fakeProps(ComponentFile)
  //
  //   const allPropsLength = Object.keys(allProps).length
  //   const requiredPropsLength = Object.keys(requiredProps).length
  //
  //   expect(allPropsLength).toBeGreaterThan(requiredPropsLength)
  // })

  it('should return an object with shape props faked', () => {
    const props = fakeProps(SimpleFile)
    const { requiredString, requiredObject } = props

    expect(props).toEqual({
      requiredString,
      requiredObject: {
        requiredString: requiredObject.requiredString
      }
    })
  })

  it('should prefix shape props', () => {
    const props = fakeProps(SimpleFile)
    expect(props.requiredObject.requiredString).toBe(
      'requiredObject.requiredString'
    )
  })
})
