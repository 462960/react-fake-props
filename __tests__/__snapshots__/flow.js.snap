// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fakeProps should return an object with all props faked (snapshot) 1`] = `
Object {
  "address": Object {
    "pincode": 1,
    "street": "address.street",
    "street2": "address.street2",
  },
  "age": 1,
  "family": Array [
    1,
  ],
  "id": 1,
  "name": "name",
  "repos": Array [
    Object {
      "demoSite": "super.demoSite",
      "npm": "super.npm",
      "url": "super.url",
    },
  ],
}
`;

exports[`fakeProps should return an object with required props faked (snapshot) 1`] = `
Object {
  "address": Object {
    "pincode": 1,
    "street": "address.street",
  },
  "age": 1,
  "id": 1,
  "name": "name",
  "repos": Array [
    Object {
      "url": "super.url",
    },
  ],
}
`;
